src/test/java/com/microsoft/sqlserver/jdbc/fedauth/PooledConnectionTest.java

```diff
@@ -37,6 +38,15 @@
     static String charTable = TestUtils.escapeSingleQuotes(
             AbstractSQLGenerator.escapeIdentifier(RandomUtil.getIdentifier("JDBC_PooledConnection")));
 
+    final AtomicReference<Throwable> throwableRef = new AtomicReference<Throwable>();
+    Thread.UncaughtExceptionHandler handler = new Thread.UncaughtExceptionHandler() {
+
+        @Override
+        public void uncaughtException(Thread t, Throwable e) {
+            throwableRef.set(e);
+        }
+    };
+

```

Comment:
Can you explain why this is necessary, why can't we just add this to the runnable catch clause?